src/ts/components/subtitleoverlay.ts

```diff
@@ -416,3 +418,82 @@ class ActiveSubtitleManager {
     this.activeSubtitleCueCount = 0;
   }
 }
+
+class SubtitleRegionContainerManager {
+  private subtitleRegionContainers: { [region: string]: SubtitleRegionContainer } = {};
+  private subtitleOverlay: SubtitleOverlay;
+
+  /**
+   * @param subtitleOverlay Reference to the subtitle overlay for adding and removing the containers.
+   */
+  constructor(subtitleOverlay: SubtitleOverlay) {
+    this.subtitleOverlay = subtitleOverlay;
+  }
+
+  /**
+   * Removes a subtitle label from a container.
+   * @param labelToRemove
+   */
+  public removeLabel(labelToRemove: SubtitleLabel): void {
+    for (let region in this.subtitleRegionContainers) {
+      this.subtitleRegionContainers[region].removeLabel(labelToRemove);
+    }
+  }
+
+  /**
+   * Removes all subtitle containers.
+   */
+  public clear(): void {
```

Comment:
Shouldn't that also be called on `cueExit`? What if there is no need for this container anymore? But I'm not sure if the `cueExitEvent` does have the needed information.

Reply:
Could there be a case where multiple subtitles are in one region and only one of the subtitles fires the `cueExit` event?

Reply:
Sounds like a default case to me.

Reply:
So it's probably better to not recreate the containers on every cueEnter / cueExit and reuse the existing ones instead.